scheduler/src/cook/mesos/task.clj

```diff
@@ -1,50 +1,57 @@
 (ns cook.mesos.task
   (:require [cook.mesos.util :as util]
             [plumbing.core :refer (map-vals)]
-            [clojure.tools.logging :as log])
+            [clojure.tools.logging :as log]
+            [cheshire.core :as json])
   (import com.netflix.fenzo.TaskAssignmentResult))
 
 (defonce custom-executor-name "cook_agent_executor")
 (defonce custom-executor-source "cook_scheduler")
 
+(defn build-task-status-data [job-ent]
+  (.getBytes
+   (if (:job/custom-executor job-ent true)
+     ;;TODO this data is a race-condition
+     (pr-str
+      {:instance (str (count (:job/instance job-ent)))})
+     (json/generate-string
+      {:commands (util/job-ent->commands job-ent)}))
+   "UTF-8"))
+
 (defn job->task-metadata
   "Takes a job entity, returns task metadata"
-  [db fid job-ent task-id]
+  [db fid executor-command job-ent task-id]
   (let [resources (util/job-ent->resources job-ent)
         ;; If the custom-executor attr isn't set, we default to using a custom
         ;; executor in order to support jobs submitted before we added this field
         container (util/job-ent->container db job-ent)
         custom-executor (:job/custom-executor job-ent true)
         environment (util/job-ent->env job-ent)
         labels (util/job-ent->label job-ent)
-        command {:value (:job/command job-ent)
+        command {:value (if custom-executor
```

Comment:
Yes, that's correct. The new executor replaces the command executor; it should be transparent to end users. This way we're able to maintain backward compatibility with existing clojure executor.